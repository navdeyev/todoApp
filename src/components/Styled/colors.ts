export enum colors {
    BLACK = '#000000',
    RED = '#F00000',
    MINE_SHAFT = '#222222',
    DOVE_GREY = '#666666',
    DUSTY_GREY = '#999999',
    ALTO = '#dddddd',
    CYAN = '#0FFFFF',
    WHITE = '#ffffff'
}
