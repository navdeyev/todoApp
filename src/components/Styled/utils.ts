import {css, SimpleInterpolation} from 'styled-components';

export const rem = (px: number): string => {
    return `${px / 16}rem`;
};

enum MediaTypes {
    MOBILE = 'only screen and (max-width: 600px)',
    TABLET = 'only screen and (min-width: 601px) and (max-width: 1023px)',
    DESKTOP = 'only screen and (min-width: 1024px)'
}

const createMediaWrapper = (mediaType: MediaTypes) =>
    (strings: TemplateStringsArray, ...interpolations: SimpleInterpolation[]) =>
        css`@media ${mediaType} { ${ css(strings, ...interpolations) } }`;

export const media = {
    mobile: createMediaWrapper(MediaTypes.MOBILE),
    tablet: createMediaWrapper(MediaTypes.TABLET),
    desktop: createMediaWrapper(MediaTypes.DESKTOP)
};