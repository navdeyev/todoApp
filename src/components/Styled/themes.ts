import * as React from 'react';
import {colors} from './colors';

export interface IHasTheme {
    children: React.ReactNode
    className?: string,
    theme?: ITheme,
}

export interface ITheme {
    pageBackgroundColor: string,
    pageMainTextColor: string,

    pageHeaderBgColor: string,
    pageTitleColor: string,

    hoverStatusBorderColor: string
    statusBorderColor: string,

    todoCardBorderColor: string
}

export const defaultTheme: ITheme = {
    pageBackgroundColor: colors.WHITE,
    pageMainTextColor: colors.BLACK,

    pageHeaderBgColor: colors.MINE_SHAFT,
    pageTitleColor: colors.WHITE,

    hoverStatusBorderColor: colors.MINE_SHAFT,
    statusBorderColor: colors.DOVE_GREY,

    todoCardBorderColor: colors.DOVE_GREY
};

export const darkTheme: ITheme = {
    pageBackgroundColor: colors.BLACK,
    pageMainTextColor: colors.WHITE,

    pageHeaderBgColor: colors.ALTO,
    pageTitleColor: colors.BLACK,

    hoverStatusBorderColor: colors.ALTO,
    statusBorderColor: colors.DUSTY_GREY,

    todoCardBorderColor: colors.DUSTY_GREY
};