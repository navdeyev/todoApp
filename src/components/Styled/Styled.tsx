import styled from 'styled-components';
import {media, rem} from './utils';

export const Heading1 = styled.h1`
    margin: 0;
    padding:0;
    font-weight: 700;
    font-size: ${ rem(24) };
    line-height: ${ rem(32) };
    
    ${ media.tablet`
        font-size: ${ rem(28) };
        line-height: ${ rem(36) };
    `}

    ${ media.desktop`
        font-size: ${ rem(32) };
        line-height: ${ rem(40) };
    `}
`;

export const Heading2 = styled.h2`
    margin: 0;
    padding:0;
    font-weight: 400;
    font-size: ${ rem(20) };
    line-height: ${ rem(28) };
    
    ${ media.tablet`
        font-size: ${ rem(22) };
        line-height: ${ rem(30) };
    `}

    ${ media.desktop`
        font-size: ${ rem(24) };
        line-height: ${ rem(32) };
    `}
`;

export const Paragraph = styled.p`
    margin: 0;
    padding:0;
    font-weight: 100;
    font-size: ${ rem(12) };
    line-height: ${ rem(16) };
    
    ${ media.tablet`
        font-size: ${ rem(14) };
        line-height: ${ rem(16) };
    `}
    
    ${ media.desktop`
        font-size: ${ rem(16) };
        line-height: ${ rem(18) };
    `}
`;