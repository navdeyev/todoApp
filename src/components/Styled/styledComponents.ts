import * as styledComponents from 'styled-components';
import {ThemedStyledComponentsModule} from 'styled-components';

import {ITheme} from './themes';

const {
    default: styled,
    css,
    injectGlobal,
    keyframes,
    ThemeProvider
} = styledComponents as ThemedStyledComponentsModule<ITheme>;

export {css, injectGlobal, keyframes, ThemeProvider};
export default styled;