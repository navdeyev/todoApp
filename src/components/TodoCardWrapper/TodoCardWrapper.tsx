import * as React from 'react';

import styled from 'components/Styled/styledComponents';
import {defaultTheme, IHasTheme} from 'components/Styled/themes';
import {rem} from 'components/Styled/utils';

const Wrapper: React.SFC<IHasTheme> = props =>
    <div className={props.className}>{props.children}</div>;

const TodoCardWrapper = styled(Wrapper)`
    width: 100%;
    box-sizing: border-box;
    border: 2px solid ${ props => props.theme.todoCardBorderColor };
    border-radius: ${ rem(6) };
    margin-bottom: ${ rem(16) };
    padding: ${ rem(20) } ${ rem(16) };
    
    &:first-child {
      margin-top: ${ rem(16) };    
    }
`;

TodoCardWrapper.defaultProps = {theme: defaultTheme};
TodoCardWrapper.displayName = 'TodoCardWrapper';

export default TodoCardWrapper;