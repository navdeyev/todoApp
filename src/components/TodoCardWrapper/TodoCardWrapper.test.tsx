import {shallow} from 'enzyme';
import * as React from 'react';

import {IHasTheme} from 'components/Styled/themes';
import TodoCardWrapper from './TodoCardWrapper';

describe('PageHeader', () => {
    it('renders a themed PageHeader component', () => {
        const props: IHasTheme = {
            children: <div>Some text here</div>
        };
        const render = shallow(<TodoCardWrapper {...props}/>);
        expect(render.getElement()).toMatchSnapshot();
    });
});
