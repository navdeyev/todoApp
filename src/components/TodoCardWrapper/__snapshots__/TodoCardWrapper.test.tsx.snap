// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader renders a themed PageHeader component 1`] = `
<Wrapper
  className="sc-bdVaJa erXgsM"
  theme={
    Object {
      "hoverStatusBorderColor": "#222222",
      "pageBackgroundColor": "#ffffff",
      "pageHeaderBgColor": "#222222",
      "pageMainTextColor": "#000000",
      "pageTitleColor": "#ffffff",
      "statusBorderColor": "#666666",
      "todoCardBorderColor": "#666666",
    }
  }
>
  <div>
    Some text here
  </div>
</Wrapper>
`;
